src/ResourceManager/CognitiveServices/Commands.Management.CognitiveServices/CognitiveServicesAccount/GetAzureCognitiveServicesAccountSkus.cs

```diff
@@ -15,45 +15,90 @@
+        public string Type { get; set; }
+
+        [Parameter(
+            ParameterSetName = GetSkusWithFilterParamSetName,
+            Position = 1,
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Cognitive Services Account Location.")]
+        [LocationCompleter("Microsoft.CognitiveServices/accounts")]
+        public string Location { get; set; }
+
         [Parameter(
+            ParameterSetName = GetSkusWithAccountParamSetName,
             Position = 0,
             Mandatory = true,
             ValueFromPipelineByPropertyName = true,
             HelpMessage = "Resource Group Name.")]
         [ResourceGroupCompleter()]
-        [ValidateNotNullOrEmpty]
         public string ResourceGroupName { get; set; }
 
         [Parameter(
+            ParameterSetName = GetSkusWithAccountParamSetName,
             Position = 1,
             Mandatory = true,
             ValueFromPipelineByPropertyName = true,
             HelpMessage = "Cognitive Services Account Name.")]
         [Alias(CognitiveServicesAccountNameAlias, AccountNameAlias)]
-        [ValidateNotNullOrEmpty]
         public string Name { get; set; }
 
         public override void ExecuteCmdlet()
         {
             base.ExecuteCmdlet();
             RunCmdLet(() =>
             {
-                var cognitiveServicesKeys = this.CognitiveServicesClient.Accounts.ListSkus(
-                     this.ResourceGroupName,
-                     this.Name);
+                if (!string.IsNullOrEmpty(this.Name))
+                {
+                    if (string.IsNullOrEmpty(this.ResourceGroupName))
+                    {
+                        throw new ErrorException("ResourceGroupName must present when Name is indicated.");
+                    }
+
+                    // keep compatibility with old behavior but give warnings.
+                    WriteWarningWithTimestamp("Get Available Skus with an existing Cognitive Services Account is deprecated and will be removed in a future version.");
+
+                    var cognitiveServicesSkus = this.CognitiveServicesClient.Accounts.ListSkus(
+                         this.ResourceGroupName,
+                         this.Name);
+
+                    WriteObject(cognitiveServicesSkus);
+                }
+                else
+                {
+                    var cognitiveServicesSkus = this.CognitiveServicesClient.ResourceSkus.List().Where(
+                        resourceSku =>
+                        (string.IsNullOrWhiteSpace(this.Type) || resourceSku.Kind == this.Type)
+                        && (string.IsNullOrWhiteSpace(this.Location) || resourceSku.Locations.Any(x => string.Compare(x, this.Location, StringComparison.OrdinalIgnoreCase) == 0))
+                    );
 
-                WriteObject(cognitiveServicesKeys);
+                    WriteObject(cognitiveServicesSkus);
```

Comment:
@yangyuan same comment about changing the call to `WriteObject(cognitiveServicesSkus, true)`

Reply:
cool, thanks.